v2/blockstore/readonly.go

```diff
@@ -22,28 +22,38 @@ import (
 	internalio "github.com/ipld/go-car/v2/internal/io"
 )
 
-var _ blockstore.Blockstore = (*ReadOnly)(nil)
-
-// ReadOnly provides a read-only CAR Block Store.
-type ReadOnly struct {
-	// mu allows ReadWrite to be safe for concurrent use.
-	// It's in ReadOnly so that read operations also grab read locks,
-	// given that ReadWrite embeds ReadOnly for methods like Get and Has.
-	//
-	// The main fields guarded by the mutex are the index and the underlying writers.
-	// For simplicity, the entirety of the blockstore methods grab the mutex.
-	mu sync.RWMutex
-
-	// The backing containing the data payload in CARv1 format.
-	backing io.ReaderAt
-	// The CARv1 content index.
-	idx index.Index
-
-	// If we called carv2.NewReaderMmap, remember to close it too.
-	carv2Closer io.Closer
-
-	ropts carv2.ReadOptions
-}
+var (
+	_                    blockstore.Blockstore = (*ReadOnly)(nil)
+	noopErrHandler                             = func(error) {}
+	errZeroLengthSection                       = fmt.Errorf("zero-length section not allowed by default; see WithZeroLengthSectionAsEOF option")
+)
+
+type (
+	// ReadOnly provides a read-only CAR Block Store.
+	ReadOnly struct {
+		// mu allows ReadWrite to be safe for concurrent use.
+		// It's in ReadOnly so that read operations also grab read locks,
+		// given that ReadWrite embeds ReadOnly for methods like Get and Has.
+		//
+		// The main fields guarded by the mutex are the index and the underlying writers.
+		// For simplicity, the entirety of the blockstore methods grab the mutex.
+		mu sync.RWMutex
+
+		// The backing containing the data payload in CARv1 format.
+		backing io.ReaderAt
+		// The CARv1 content index.
+		idx index.Index
+
+		// If we called carv2.NewReaderMmap, remember to close it too.
+		carv2Closer io.Closer
+
+		ropts carv2.ReadOptions
+	}
+
+	contextKey int
+)
+
+const asyncErrHandlerKey contextKey = iota
```

Comment:
why use iota at all, if this is just one value?

Reply:
The rationale was: it is one value today. Don't know if we would add to this or not in the future. Will make it zero.

Reply:
It used to be recommended that context value keys be strings to avoid allocations. I'm not sure what the status is but this issue seems to suggest that strings are still preferred: https://github.com/golang/go/issues/33742

Reply:
Thank you @iand for pointing this out; will apply 🍻